File: sdk/dotnet-sdk/src/VeritySDK.Test/Protocols/MessageFamilyTest.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using VeritySDK.Exceptions;
using VeritySDK.Protocols;
using VeritySDK.Transports;
using VeritySDK.Utils;

namespace VeritySDK.Test
{
    [TestClass]
    public class MessageFamilyTest
    {
        internal class TestFamily : MessageFamily
        {
            public override string qualifier() { return Util.EVERNYM_MSG_QUALIFIER; }
            public override string family() { return "testing"; }
            public override string version() { return "0.1"; }
        };

        private TestFamily testFamily = new TestFamily();

        [TestMethod]
        public void extractMessageName()
        {
            string name = testFamily.messageName("did:sov:123456789abcdefghi1234;spec/testing/0.1/test");
            Assert.AreEqual("test", name);

            name = testFamily.messageName("did:sov:123456789abcdefghi1234;spec/testing/0.1/234asdf234@$");
            Assert.AreEqual("234asdf234@$", name);
        }

        [TestMethod]
        public void invalidMessageFamily()
        {
            Assert.ThrowsException<InvalidMessageTypeException>(() =>
                testFamily.messageName("did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/testing/0.1/test")
            );
        }

        [TestMethod]
        public void invalidMessageName()
        {
            Assert.ThrowsException<InvalidMessageTypeException>(() =>
                testFamily.messageName("did:sov:123456789abcdefghi1234;spec/testing/0.1/")
            );
        }

        [TestMethod]
        public void invalidMessageName2()
        {
            Assert.ThrowsException<InvalidMessageTypeException>(() =>
                testFamily.messageName("did:sov:123456789abcdefghi1234;spec/testing/0.1-test")
            );
        }
    }
}

File: sdk/java-sdk/src/test/java/com/evernym/verity/sdk/protocols/MessageFamilyTest.java
package com.evernym.verity.sdk.protocols;

import com.evernym.verity.sdk.exceptions.InvalidMessageTypeException;
import com.evernym.verity.sdk.utils.Util;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class MessageFamilyTest {
    public final static MessageFamily testFamily = new MessageFamily(){
        @Override public String qualifier() {return Util.EVERNYM_MSG_QUALIFIER;}
        @Override public String family() {return "testing";}
        @Override public String version() {return "0.1";}
    };

    @Test
    public void extractMessageName() throws InvalidMessageTypeException {
        String name = testFamily.messageName(Util.EVERNYM_MSG_QUALIFIER + "/testing/0.1/test");
        assertEquals("test", name);

        name = testFamily.messageName(Util.EVERNYM_MSG_QUALIFIER + "/testing/0.1/234asdf234@$");
        assertEquals("234asdf234@$", name);
    }

    @Test
    public void invalidMessageFamily() throws InvalidMessageTypeException {
        Assertions.assertThrows(InvalidMessageTypeException.class, () -> {
            testFamily.messageName(Util.COMMUNITY_MSG_QUALIFIER + "/testing/0.1/test");
        });
    }

    @Test
    public void invalidMessageName() throws InvalidMessageTypeException {
        Assertions.assertThrows(InvalidMessageTypeException.class, () -> {
            testFamily.messageName(Util.EVERNYM_MSG_QUALIFIER + "/testing/0.1/");
        });
    }

    @Test
    public void invalidMessageName2() throws InvalidMessageTypeException {
        Assertions.assertThrows(InvalidMessageTypeException.class, () -> {
            testFamily.messageName(Util.EVERNYM_MSG_QUALIFIER + "c/testing/0.1-test");
        });
    }
}
